src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1148,7 +1148,9 @@ public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
             for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.FileKeyword; i++)
             {
-                yield return (SyntaxKind)i;
+                // 8441 corresponds to a deleted kind (DataKeyword) that was previously shipped.
```

Comment:
Maybe add `Debug.Assert(((SyntaxKind)i).ToString() != i.ToString())`

Reply:
Simplified your suggestion to `Debug.Assert(Enum.IsDefined(typeof(SyntaxKind), (SyntaxKind)i))` (had to use version with `typeof` due to the fact, that generic overload is not present in `netstandard2.0`). Does the same things, but reads a lot easier. Also since there are multiple methods, which loops through numbers like this added such guard to all of them. Not sure, whether this will be classified as an "unrelated" change by the compiler team, but since we faced bug of this nature, I guess, it's better to add such guard rails everywhere

Reply:
Adding additional asserts revealed that are a lot unexisting kinds are returned from `GetPunctuationKinds()` method